### ISSUE TITLE
[BUG] Can't load a file from documents directory into the webview / '_file_' prefix causes SIGBART on iOS 

### ISSUE THREAD

**Issue by JRR-OSU** [ionic-team/cordova-plugin-ionic-webview#133]
I recently uncovered a treasure trove of bugs with this plugin after migrating to 2.0. Previously we were on 1.1.19 and had no issues loading a local audio file into a waveform on iOS. For my current project we need to update our plugin to > 2.0 to be able to interface with Ionic Pro, however this completely broke loading in local files into the waveform container on iOS.

This is the line that does it: ` this.wavesurfer.load('_file_' + this.file.documentsDirectory.replace(/file:\/\//g, '')+ fileName);`

Now previously, on 1.1.19, this line worked as, but this does not work on 2.0:  `this.wavesurfer.load(this.file.documentsDirectory.replace(/file:\/\//g, '')+ fileName;`

I've tried using `'_file_'`, `"file://"` and `"cdvfile://"`. None work.

When trying with the `'_file_'` prefix, I get a SIGBART on iOS.

![image](https://user-images.githubusercontent.com/12603147/43361206-0eabe1f0-9297-11e8-9ad2-9753ab045ea3.png)

Bug 1: `self.CDV_LOCAL_SERVER` is getting set to nil somewhere
Bug 2: If I hard code the local server path, I get a SIGBART deep inside the engine code. Can post more on that later.

Is there something I'm missing here? I get that the webview now runs inside the local server with a different path, but how can i adjust my code to be able to load my waveform??

Help is needed urgently, thanks.

**Comment by scottmizo** [comment]
I'm running into the same issue with 2.0.1 when using Ionic.WebView.convertFileSrc() to generate a URL from a file entry.

To my novice eyes, it appears the file handler is getting set prior to CDV_LOCAL_SERVER bing defined.

workaround - *serverUrl = @"http://localhost:8080"

**Comment by JRR-OSU** [comment]
This is the result I get using the workaround: 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/12603147/43369828-b1ea0ed6-9342-11e8-87a5-813c1060cb49.png">

**Comment by JRR-OSU** [comment]
This is the result I get when using the workaround... 

![screen shot 2018-07-29 at 3 19 27 pm](https://user-images.githubusercontent.com/12603147/43369843-de1563ac-9342-11e8-8f90-400e8c90a417.png)

**Comment by matejkramny** [comment]
@JRR-OSU that means the file is missing from the filesystem

**Comment by matejkramny** [comment]
Sorry messed up the merge request. Present now on #135

**Comment by olivermuc** [comment]
I'm seeing the same issues with accessing local files since the upgrade.
In the previous 1x version, a mere "file://" (or not for iOS) would do the trick.

I simply updated to cordova-plugin-ionic-webview v2 and it broke local file access for Android:
I tried both hardcoding the server address and using 
`window.Ionic.WebView.convertFileSrc(filePath);`
both produce the same result in fact, eg:
`http://localhost:8080/_file_/data/user/0/com....some.app/files/assets/e6d1ac1d14b6f15f1124dab9e7488ff8"`

I don't see any security/access related errors but also nothing shows, leaving images broken.

On iOS it is even worse, I too see the dreaded "*signal SIGABRT*" error which leads to crashing the app:
```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString replaceOccurrencesOfString:withString:options:range:]: nil argument'
*** First throw call stack:
(0x181afed8c 0x180cb85ec 0x181afec6c 0x181a0c1a8 0x18245fe58 0x100990c60 0x1009a3e34 0x10099d274 0x10099f8f4 0x1009a0ee0 0x1009a0738 0x10112919c 0x1011011dc 0x10110119c 0x101112a00 0x101112500 0x181723fac 0x181723b08)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by matejkramny** [comment]
@olivermuc try patching `CDVWKWebViewEngine.m` with my MR

**Comment by olivermuc** [comment]
Excellent @matejkramny - I sincerely hope we can get this fix in asap. Awesome catch!

**Comment by tetpsy** [comment]
@matejkramny Your MR worked perfectly for me. Thanks

**Comment by JRR-OSU** [comment]
@matejkramny Thanks for posting a fix. I have no idea why the person that wrote this decided to put nil values everywhere with no nil checks. There are other issues with the range checks for <audio> and <video> tags in the engine code as well. Hopefully over time we can get these sorted!

**Comment by Mapiac** [comment]
Wondering @matejkramny did you find a fix w Android? Or maybe @JRR-OSU or @olivermuc did you guys have Android issues on latest WKWebView?

**Comment by JRR-OSU** [comment]
@Mapiac I didn't have issues on Android simply because I went with Crosswalk to avoid issues like this. Also I think Android avoided these issues altogether in the way that it handles the HTTP server. Previously the Ionic team used essentially a webserver plugin written by someone else on iOS that clearly didn't handle things like range requests properly. I think in the latest iterations they got rid of it. I needed to support devices back to iOS 9, so I just ended up fixing bugs myself. Was tedious as hell. Once I was able to get Crosswalk to build on Android, I've since had zero issues.
